[Xilef121] iP

Xilef121 opened the pull request:
### Xilef
Your one stop approach to managing all your tasks. It allows you to,

- Keep track of tasks
- Mark when your tasks are complete
- Ensure your tasks do not clash with each other

All you need to do is, 

1. Sign up [here](https://github.com/nus-cs2103-AY2324S2/ip/pull/379)
2. Download
3. Run

And it is **FREE!**

Features:

- [x] Manage Tasks
- [ ] Manage deadlines
- [ ] Reminders

If you are a programmer as well and you wish to write your own code, use this as a template:
```
public Duke() {
        ui = new Ui();
        storage = new Storage(FILE_PATH);
        try {
            tasks = new TaskList(storage.load());
        } catch (DukeException e) {
            ui.showLoadingError();
            tasks = new TaskList();
        }
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Xilef121 committed:
Changed name, added greetings and exiting.

Xilef121 committed:
Added echo and bye bye.

Xilef121 committed:
Added add and list

Xilef121 committed:
Added a Task class, added methods to task and utilised them in Duke for marking and unmarking

Xilef121 committed:
Added Deadline, Event, Todo classes. Changed from using susbtring() to split() and implemented a new toString method in Task

Xilef121 committed:
Made changes to some of the strings, whitespaces, lines and using automated text ui testing.

Xilef121 committed:
Added DukeException class, implemented some error catching

Xilef121 committed:
Added delete function and added some error catching for list and delete

Xilef121 committed:
Add new methods in Duke to read and write files

Xilef121 committed:
Add date understanding
Add new date representation format

Xilef121 committed:
Add error checking

Xilef121 committed:
Add time to deadlines and events

fy17ohhh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,195 @@
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
```
Perhaps fix the import statements to follow lexicographical order? e.g. `java.io.File` should be before `java.nio.file.Paths`

fy17ohhh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,195 @@
+                t = new Event(taskDescription, taskTo,  taskFrom);
+                break;
+            default:
+                System.out.println("Invalid task type: " + taskType);
```
Should there be no indentation for `case` clauses?

fy17ohhh commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,23 @@
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + " (from: " + from.format(DateTimeFormatter.ofPattern("d MMM yyyy| hh:mm a")) +
+                " to: " + to.format(DateTimeFormatter.ofPattern("d MMM yyyy| hh:mm a")) + ")";
```
Should there be break before an operator instead of break after an operator?
```suggestion
        return "[E]" + super.toString() + " (from: " + from.format(DateTimeFormatter.ofPattern("d MMM yyyy| hh:mm a"))
                + " to: " + to.format(DateTimeFormatter.ofPattern("d MMM yyyy| hh:mm a")) + ")";
```

fy17ohhh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,195 @@
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        Scanner scanner = new Scanner(System.in);
+        ArrayList<Task> list = loadFile();
```
Would it be better to use a more meaningful variable name instead?

fy17ohhh commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,35 @@
+            d = "1";
+        } else {
+            d = "0";
+        }
```
Perhaps use ternary operator to make code more tidy 🤔

fy17ohhh reviewed (commented):
Overall, I found your code well organised and clear to read. Just a few nits in coding standards to fix.

Xilef121 committed:
Add new classes for better OOP

Xilef121 committed:
Merge branch 'branch-MoreOOP'

* branch-MoreOOP:
  Add new classes for better OOP

Xilef121 committed:
Add duke package, duke.task package, duke.command package

Xilef121 committed:
Merge branch 'branch-Packages'

* branch-Packages:
  Add duke package, duke.task package, duke.command package

Xilef121 committed:
Merge commit 'a6f73244a20539e20c2189973ebf597df34ef631'

* commit 'a6f73244a20539e20c2189973ebf597df34ef631':
  Bump gradle and lib version
  Add Gradle support
  docs/README.md: Tweak document template

Xilef121 committed:
Automate project build using Gradle

Xilef121 committed:
Add JUnit tests for Parser and Todo class

Xilef121 committed:
Merge branch 'branch-JUnit'

* branch-JUnit:
  Add JUnit tests for Parser and Todo class

Xilef121 committed:
Create duke.jar

Xilef121 committed:
Add javadoc

Xilef121 committed:
Add FindCommand class

Xilef121 committed:
Edit class to follow coding standard

Xilef121 committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Add javadoc

Xilef121 committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Edit class to follow coding standard

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Storage.java
#	src/main/java/duke/command/AddCommand.java
#	src/main/java/duke/command/Command.java
#	src/main/java/duke/command/DeleteCommand.java
#	src/main/java/duke/command/ListCommand.java
#	src/main/java/duke/command/UnmarkCommand.java
#	src/main/java/duke/task/Deadline.java
#	src/main/java/duke/task/Event.java
#	src/main/java/duke/task/Todo.java

Xilef121 committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Add FindCommand class

# Conflicts:
#	src/main/java/duke/task/Task.java

Xilef121 committed:
Resolved merge conflicts

yuechen2001 commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,55 @@
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("yyyy-dd-MM HHmm");
+
+    @Test
+    public void parse_byeCommand_returnsExitCommand() throws DukeException {
```
I like how you name your test methods, it is very clear to me what you are testing for and what is the expected result.

yuechen2001 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,107 @@
+     */
+    private static Task createTaskFromLine(String line) {
+        Task t = null;
+        String[] parts = line.split("\\|");
```
Very clear naming of variables!

yuechen2001 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,64 @@
+    /**
+     * Marks the {@code Task} as not done.
+     */
+    public void unmarkDone() {
```
Personally I felt that unmarkDone is not a very intuitive name, perhaps just using unmark will be better?

yuechen2001 reviewed (commented):
Good job overall, could hardly find anything to comment on for you to improve :D!

Xilef121 committed:
Add GUI to project

Xilef121 committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Add GUI to project

Xilef121 committed:
Add assertions to Parser class and Command class

Xilef121 committed:
Fix code to follow coding standard

Xilef121 committed:
Merge pull request #2 from Xilef121/branch-A-Assertions

Add assertions to Parser class and Command class

Xilef121 committed:
Merge branch 'master' into branch-A-CodeQuality

Xilef121 committed:
Merge pull request #3 from Xilef121/branch-A-CodeQuality

Fix code to follow coding standard

Xilef121 committed:
Merge branch 'master' of ssh://github.com/Xilef121/ip

* 'master' of ssh://github.com/Xilef121/ip:
  Fix code to follow coding standard
  Add assertions to Parser class and Command class

Xilef121 committed:
Implement Undo function to chatbot

The current version of the chatbot lacked the capability to undo user
interactions, leading to potential user frustration and limited
usability.

To address this issue, I have implemented an undo function within the
chatbot system.

Let's,
*implement a stack-based mechanism to track user inputs and commands
*introduce a UndoCommand class to handle reverting user actions
*ensure seamless integration with existing command funcitonalities

Xilef121 committed:
Merge pull request #4 from Xilef121/branch-C-Undo

Implement Undo function to chatbot

Xilef121 committed:
Merge branch 'master' of ssh://github.com/Xilef121/ip

* 'master' of ssh://github.com/Xilef121/ip:
  Implement Undo function to chatbot

Xilef121 committed:
Refactor Duke into Xilef

Xilef121 committed:
Merge branch 'branch-Refactoring'

* branch-Refactoring:
  Refactor Duke into Xilef

Xilef121 committed:
Refactor leftover traces of duke and improve GUI images

Xilef121 committed:
Add javadocs for new classes and add return for command classes

Xilef121 committed:
Fix undo functionality and text wrapping in GUI

The undo functionality in the Undo class was not properly restoring
deleted tasks to their original positions in the task list.
Additionally, the text wrapping feature in the GUI was not handling long
texts properly.

The improper undo functionality could lead to inconsistencies in
task management, while the lack of text wrapping affects the user
experience negatively.

Let's,
*update the undo class to insert the task back into the proper position
*modify the dialogbox.fxml file to implement proper text wrapping for
the gui

Xilef121 committed:
Update user guide for Xilef and upload product screenshot

Xilef121 committed:
Fix storage to create a data directory if not found
